packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -339,7 +374,8 @@ function FlowsheetDataGrid(props) {
       if (anchorCell.current === null && !gridRef.rows[rowIndex].hasAttribute('data-section-id')) {
         anchorCell.current = {
           rowId: gridRef.rows[rowIndex].getAttribute('data-row-id'),
-          columnId: columns[columnIndex].id,
+          columnId: displayedColumnsWithColumnSpan[columnIndex].id,
+          columnIndex,
```

Comment:
Why was this change required?

Reply:
If you are asking about why columnIndex was added, the cell selection logic won't work based off of just the column ID and row ID anymore because there is no more 1-1 mapping between cells and column IDs with column spans. We need the column Index for cell selection to work properly.